src/transformers/modeling_blenderbot.py

```diff
@@ -0,0 +1,582 @@
+import math
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from torch.nn import LayerNorm
+from transformers.modeling_utils import PreTrainedModel
+from configuration_blenderbot import BlenderbotConfig
+import numpy as np
+import warnings
+
+BLENDERBOT_PRETRAINED_MODEL_ARCHIVE_MAP = {"blenderbot": "https://cdn.huggingface.co/"}
+
+
+class BlenderbotMultiHeadAttention(nn.Module):
+    """
+    multiheadattention from vaswani paper
+    """
+
+    def __init__(self, n_heads, dim, dropout=0.0):
```

Comment:
pass the whole config here and extract `n_heads` and `dim` (which I would rename to `hidden_size` as in Bert).
That's the usual way it's done and makes the classes more readable :-)

Reply:
got it